### PR TITLE
Fixed an issue where some browser or system shortcuts left the CTRL/CMD key held down

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

Currently, most of the keyboard events is handled by the `ShortcutManager`. Internal recorder is responsible for catching `keydown` and `keyup` events. Unfortunately, using browser or system shortcut breaks a flow which releasing list of pressed keys. After triggering such a shortcut, a `keyup` event isn't emitted.

I'm afraid that there is no alternative way of fixing the **source of the problem**. This solution fixes just found side effects by adding extra callback to some hook. It also changing helpers for tests. Separately used jQuery simulate events don't transfer information about pressed meta keys to each other. I decided not to pass extra options to every method call, but use modified helper.

**Note**: I haven't added extra e2e test as is seems to be impossible to trigger system/browser events using simulated events.

### How has this been tested?

I was testing cases from the issue and other system/browser shortcuts.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. [#31](https://github.com/handsontable/dev-handsontable/issues/31)

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable/src/plugins/multiColumnSorting/__tests__/multiColumnSorting.spec.js [hunk 1]
```diff
@@ -13,9 +13,7 @@ describe('MultiColumnSorting', () => {
         throw Error('Please check the test scenario. The header doesn\'t exist.');
       }
 
-      $spanInsideHeader.simulate('mousedown');
-      $spanInsideHeader.simulate('mouseup');
-      $spanInsideHeader.simulate('click');
+      simulateClick($spanInsideHeader[0]);
```
Comment (budnix) [inline handsontable/src/plugins/multiColumnSorting/__tests__/multiColumnSorting.spec.js:16]:
```suggestion
      simulateClick($spanInsideHeader);
```
AFAIK the helper accepts the jQuery element instance as well.

File: handsontable/src/plugins/columnSorting/__tests__/columnSorting.spec.js [hunk 2]
```diff
@@ -13,9 +13,7 @@ describe('ColumnSorting', () => {
         throw Error('Please check the test scenario. The header doesn\'t exist.');
       }
 
-      $spanInsideHeader.simulate('mousedown');
-      $spanInsideHeader.simulate('mouseup');
-      $spanInsideHeader.simulate('click');
+      simulateClick($spanInsideHeader[0]);
```
Comment (budnix) [inline handsontable/src/plugins/columnSorting/__tests__/columnSorting.spec.js:16]:
```suggestion
      simulateClick($spanInsideHeader);
```
AFAIK the helper accepts the jQuery element instance as well.

**Comment by AMBudnik** [comment]
I reported @budnix issue here https://github.com/handsontable/dev-handsontable/issues/1025 as a new case, as all of the reported cases at https://github.com/handsontable/dev-handsontable/issues/31 relates to lack of CTRL/CMD release and this case relates to the opposite - lack of CTRL/CMD activation for range cell selection.

For the original *Case 3*, we got the video reproduction from the user

https://user-images.githubusercontent.com/10757813/212658067-092c4ca2-a0c3-47c7-a107-72a346e8ae38.mov

and the following steps:
1. press ctrl
2. click on a few cells
3. release ctrl
4. click on a few cells
5. click outside of table
6. click on a few cells

**Users environment: macOS/Chrome 78**

File: handsontable/src/plugins/columnSorting/__tests__/columnSorting.spec.js (hunk 2 above)
Comment (wszymanski) [reply handsontable/src/plugins/columnSorting/__tests__/columnSorting.spec.js]: Changed within https://github.com/handsontable/handsontable/pull/10210/commits/5df751b593920ca0f2e63f09e7a9522ec8d0425c.

File: handsontable/src/plugins/multiColumnSorting/__tests__/multiColumnSorting.spec.js (hunk 1 above)
Comment (wszymanski) [reply handsontable/src/plugins/multiColumnSorting/__tests__/multiColumnSorting.spec.js]: Changed within https://github.com/handsontable/handsontable/pull/10210/commits/5df751b593920ca0f2e63f09e7a9522ec8d0425c.

File: handsontable/src/shortcuts/__tests__/shortcutManager.spec.js [hunk 3]
```diff
@@ -478,4 +478,23 @@ describe('shortcutManager', () => {
 
     document.body.removeChild(externalInputElement);
   });
+
+  it('should check if there is a need of releasing keys on click #32', () => {
```
Comment (budnix) [inline handsontable/src/shortcuts/__tests__/shortcutManager.spec.js:482]:
```suggestion
  it('should check if there is a need of releasing keys on click #dev-1025', () => {
```
`#32`? Here shouldn't be [#1025 ](https://github.com/handsontable/dev-handsontable/issues/1025)?

**Review by budnix — CHANGES_REQUESTED** [review]
Looks good 👍 There is one suggestion left to resolve before merging.

**Review by budnix — APPROVED** [review]

**Comment by aninde** [comment]
I can confirm that the reported bug has been fixed on all supported environments. I also checked it by switching focus between Handsontable and an HTML input (https://jsfiddle.net/aninde/5rp0g72y/) and a text area. I can confirm that other keyboard shortcuts, such as those using control, meta, and alt, are still working properly.

**Comment by AMBudnik** [comment]
Released in Handsontable v12.3.1

**Pull request merged**